lib/galaxy/managers/genomes.py

```diff
@@ -0,0 +1,78 @@
+from typing import (
+    Any,
+    List,
+)
+
+from galaxy import model as m
+from galaxy.exceptions import (
+    ReferenceDataError,
+    RequestParameterInvalidException,
+)
+from galaxy.managers.context import ProvidesUserContext
+from galaxy.structured_app import StructuredApp
+
+
+class GenomesManager:
+
+    def __init__(self, app: StructuredApp):
+        self._app = app
+        self.genomes = app.genomes
+
+    def get_dbkeys(self, user: m.User, chrom_info: bool) -> List[List[str]]:
+        return self.genomes.get_dbkeys(user, chrom_info)
+
+    def get_genome(
+        self,
+        trans: ProvidesUserContext,
+        id: str,
+        num: int,
+        chrom: str,
+        low: int,
+        high: int,
+        reference: bool
+    ) -> Any:
+        if reference:
+            region = self.genomes.reference(trans, dbkey=id, chrom=chrom, low=low, high=high)
+            return {'dataset_type': 'refseq', 'data': region.sequence}
+        else:
+            return self.genomes.chroms(trans, dbkey=id, num=num, chrom=chrom, low=low)
+
+    def get_sequence(
+        self,
+        trans: ProvidesUserContext,
+        id: str,
+        chrom: str,
+        low: int,
+        high: int
+    ) -> Any:
+        region = self.genomes.reference(trans, dbkey=id, chrom=chrom, low=low, high=high)
+        return region.sequence
+
+    def get_indexes(self, id: str, index_type: str) -> Any:
+        index_extensions = {'fasta_indexes': '.fai'}
+        if index_type not in index_extensions:
+            raise RequestParameterInvalidException(f'Invalid index type: {index_type}')
+
+        tbl_entries = self._app.tool_data_tables.data_tables[index_type].data
+        ext = index_extensions[index_type]
+        index_filename = self._get_index_filename(id, tbl_entries, ext, index_type)
+
+        try:
+            fh = open(index_filename, mode='r')
+        except OSError:
+            raise ReferenceDataError(f'Failed to load index file for {id}')
+        else:
+            return fh.read()
+        finally:
+            fh.close()
```

Comment:
Can you use a with statement here ? 

I'd also say this should just return a open file handle so our middleware can do the appropriate type of serving, BUT, I don't think this is worth doing for this endpoint in this PR. There are other endpoint though that must do this. I'm not sure how that will look like, but https://github.com/mvdbeek/galaxy/blob/12d27397659b28902e4aee2ab7f668fbaf160197/lib/galaxy/web/framework/base.py#L484 is how we do this in the old wsgi part.

Reply:
Made it a with statement. Thanks for the reference to the code - I'll use it for other endpoints (which I can more easily test).